src/utils/IntlWrapper.ts

```diff
@@ -0,0 +1,412 @@
+  weeks: 7 * 24 * 60 * 60 * 1000,
+  days: 24 * 60 * 60 * 1000,
+  hours: 60 * 60 * 1000,
+  minutes: 60 * 1000,
+  seconds: 1000,
+  milliseconds: 1,
+} as const;
+
+export class IntlDuration {
+  protected duration?: DurationFormat;
+
+  protected locale: Intl.LocalesArgument;
+
+  // @ts-expect-error surely it works
+  protected isFallback = !Intl.DurationFormat;
+
+  constructor(locale: Intl.LocalesArgument = api.storage.lang('locale')) {
+    this.locale = locale;
+    return this;
+  }
+
+  setLocale(locale: Intl.LocalesArgument) {
+    this.locale = locale;
+    return this;
+  }
+
+  // For {hours: 2, minutes: 15}
+  setDuration(duration: DurationFormat) {
+    this.duration = duration;
+    return this;
+  }
+
+  // For {minutes: 155}
+  setDurationFormatted(duration: DurationFormat, style: DurationStyle = 'Duration') {
+    const ms = IntlDuration.durationToMs(duration);
+    this.setRelativeTime(ms, 'milliseconds', style);
+    return this;
+  }
+
+  // For 9000 (seconds)
+  setRelativeTime(
+    relativeTime: number,
+    convertFrom: keyof typeof dateUnitToMs,
+    style: DurationStyle = 'Duration',
+  ) {
+    const ms = relativeTime * dateUnitToMs[convertFrom];
+    this.duration = this.process(ms, style);
+    return this;
+  }
+
+  // For 1733320467580 (timestamp)
+  setTimestamp(
+    timestamp: number,
+    style: DurationStyle = 'Duration',
+    relativeTo: Date | number = new Date(),
+  ) {
+    const ms = timestamp - new Date(relativeTo).getTime();
+    this.setRelativeTime(ms, 'milliseconds', style);
+    return this;
+  }
+
+  protected process(relativeTime: number, style: DurationStyle) {
+    switch (style) {
+      case 'Duration':
+        return relativeToDuration(relativeTime, ['seconds', 'minutes', 'hours', 'days']).duration;
+      case 'Progress':
+        if (!relativeTime) break;
+        const time = relativeToDuration(relativeTime, [
+          'seconds',
+          'minutes',
+          'hours',
+          'days',
+          'years',
+        ]);
+        return shortTime(time.duration);
+      case 'M/H/D/Y':
+        if (!relativeTime) break;
+        return relativeToDuration(relativeTime, ['minutes', 'hours', 'days', 'years']).duration;
+      default:
+        return {};
+    }
+    return {};
+  }
+
+  static durationToMs(input: DurationFormat): number {
+    const keys = Object.keys(input);
+    let timestamp = 0;
+    if (keys.length <= 0) return timestamp;
+    for (let i = 0; i < keys.length; i++) {
+      const key = keys[i];
+      const value = input[key];
+      timestamp += dateUnitToMs[key] * value;
+    }
+    return timestamp;
+  }
+
+  getRelativeText(options: DurationFormatOptions = { style: 'narrow' }): string {
+    if (!this.duration) return '';
+    if (this.isFallback) return timeToString(this.duration);
+    const newOptions: DurationFormatOptions = options;
+    if (options.style === 'significantLongNarrow') {
+      newOptions[Object.keys(this.duration)[0]] = 'long';
```

Comment:
It is really easy to get around it with `Object.keys({days: 2, years: 2})[0]`

I think the better idea would be to use dateUnitToMs.find() so that it iterates from years down until it finds one entry.

Reply:
oh... didn't thought about that case

Reply:
There's no `find` so just checking existence of key